src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -43,12 +51,115 @@ private boolean hasNotUpStationAndDownStation(Section newSection) {
         return !stations.contains(newSection.getUpStation()) && !stations.contains(newSection.getDownStation());
     }
 
-    public List<Station> findAllStations() {
+    public void delete(Station station) throws NotFoundStationException {
+        validateDelete(station);
+
+        Section firstSection = findFirstSection();
+        Section lastSection = findLastSection();
+        if (firstSection.getUpStation().equals(station)) {
+            deleteFirstOrLastSection(firstSection);
+            return;
+        }
+        if (lastSection.getDownStation().equals(station)) {
+            deleteFirstOrLastSection(lastSection);
+            return;
+        }
+
+        deleteMiddleSection(station);
+    }
+
+    private void validateDelete(Station station) throws NotFoundStationException {
+        if (!findAllStations().contains(station)) {
+            throw new NotFoundStationException(station.getId());
+        }
+        if (sections.size() <= MIN_SECTION_SIZE) {
+            throw new IllegalArgumentException(ERROR_MESSAGE_MIN_SECTION_SIZE);
+        }
+    }
+
+    private Section findFirstSection() {
         return sections.stream()
+                .filter(section -> isEmptyDownStation(section.getUpStation()))
+                .findAny()
+                .orElseThrow(() -> new NoSuchElementException(ERROR_MESSAGE_NO_FIRST_SECTION));
+    }
+
+    private boolean isEmptyDownStation(Station upStation) {
+        int equalCount = 0;
+        for (Station downStation : getDownStations()) {
+            equalCount = matchEqaulCount(equalCount, upStation, downStation);
+        }
+        return equalCount == 0;
+    }
+
+
+    private List<Station> getDownStations() {
+        return Collections.unmodifiableList(sections.stream()
+                .map(Section::getDownStation)
+                .collect(Collectors.toList()));
+    }
+
+    private Section findLastSection() {
+        return sections.stream()
+                .filter(section -> isEmptyUpStation(section.getDownStation()))
+                .findAny()
+                .orElseThrow(() -> new NoSuchElementException(ERROR_MESSAGE_NO_LAST_SECTION));
+    }
+
+    private boolean isEmptyUpStation(Station downStation) {
+        int equalCount = 0;
+        for (Station upStation : getUpStations()) {
+            equalCount = matchEqaulCount(equalCount, upStation, downStation);
+        }
```

Comment:
List<Station> upSations = getUpStations();
```suggestion
        for (Station upStation : upSations) {
            equalCount = matchEqaulCount(equalCount, upStation, downStation);
        }
```